### ISSUE TITLE
Critical Dr.Jit compiler failure: jit_optix_check(): API error 7012 (OPTIX_ERROR_DISK_CACHE_DATABASE_ERROR): "Disk cache database error" in /project/ext/drjit-core/src/optix_core.cpp:70.

### ISSUE THREAD

**Issue by n-kubiak** [mitsuba-renderer/drjit#130]
Hi,

I keep getting this error. Previously I would see this on HPC when 2+ jobs landed on the same server ([known issue](https://github.com/mitsuba-renderer/drjit/issues/115)) but now it also started to happen on my personal Linux machine or when a single job is running on a given HPC server. I've tried updating to the latest version of mitsuba/drjit and deleting the contents of my .drjit folder, but it didn't help. Any advice? 

Critical Dr.Jit compiler failure: jit_optix_check(): API error 7012 (OPTIX_ERROR_DISK_CACHE_DATABASE_ERROR): "Disk cache database error" in /project/ext/drjit-core/src/optix_core.cpp:70.

Many thanks,
NK

**Comment by njroussel** [comment]
Hi @n-kubiak

This is rather surprising, here's the [OptiX documentation](https://raytracing-docs.nvidia.com/optix7/api/optix__7__host_8h.html#a59a60f5f600df0f9321b0a0b1090d76b) for that function call.

Nothing jumpy to mind, could you double check the function arguments to that call ? This would require inserting prints and compiling the  project yourself.

**Comment by n-kubiak** [comment]
I've realised there is optix7cache.db in .drjit and /var/tmp/OptixCache_<username> - what's the difference between these?

**Comment by njroussel** [comment]
I don't think Dr.Jit should be using the `/var/tmp` location. As per the documentation, we should be relocating the cache folder. Maybe you have some other OptiX applications which are using the default cache location.

**Comment by n-kubiak** [comment]
I've realised the cache isn't stored on a specific machine but rather in my user space... which is accessible from any university machine I use, be it my personal Linux machine or one on the HPC cluster. So that explains the multiple access problem...

In the end, I decided not to rebuild Mitsuba from source and instead I overwrite $HOME for every experiment I run and point it towards a folder specific to a given run. A bit hacky, but it works!
